### ISSUE TITLE
About focal_loss

### ISSUE THREAD

**Issue by s7ev3n** [holyseven/PSPNet-TF-Reproduce#13]
Hi, 
Thanks for sharing your work!
I noticed you use focal loss in your code, and use 'reduce_sum' returning the loss. And I also use focal loss in my code with a little modification using 'reduce_mean' to calculate loss value, since using 'reduce_sum' output a huge loss value. However, using 'reduce_mean' also has side effect when I use pretrained variables on coarse to further train on fine labels, it start to output nan in the middle of training. 
Shall you share your insight on this issue, thanks so much!!

-s7ev3n

**Comment by holyseven** [comment]
Maybe this is caused by numerical instability. Gradients of cross entropy with softmax are simple: `-yi + pi`; but gradients of focal loss are involved with `ln pi`. 
In fact, I tested `focal_loss` with SDB dataset and used 20 examples in a mini-batch, and didn't find the `nan` problem. The results are similar with normal loss. But 8 examples with SDB, or with Cityscapes, always `nan`. 

And `reduce_sum` should be used here because I divided the total loss by a [number of valid pixels](https://github.com/holyseven/PSPNet-TF-Reproduce/blob/fb9ae4815c2df221e51b29e2083a1b6550bafcbf/model/pspnet_mg.py#L313-L319).
